### PR TITLE
enh(purge) Adds audit log purge

### PR DESCRIPTION
**Pull request by tanguyvda**
<h1> Adds audit log purge log </h1>

<h2> allows people to configure data retention for audit logs </h2>

This adds a new field in the form in administration -> parameters -> options at the bottom of the page. Field that allows people to configure the retention duration for their audit log. It also displays logs from CLAPI actions

<h2> Type of change </h2>

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

<h2> Target serie </h2>

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

-head over the administration -> parameters -> options and set a value for your audit log retention duration.

-create a backup of your database so you'll be able to repeat the next step as much as you want

-mysqldump -u centreon -p centreon_storage log_action log_action_modification > /tmp/backup_audit_log_tables.sql

In my exemple, I have the following logs
https://user-images.githubusercontent.com/7352865/56137468-38d8aa00-5f95-11e9-8b80-61f221df8fed.png

I configure my audit logs retention to 3 days so most of my past actions are going to be purged
https://user-images.githubusercontent.com/7352865/56137519-586fd280-5f95-11e9-86b9-a6df66013c2a.png

if you've done your backup, you're now ready to use the purge

write the following command on your central server
/opt/rh/rh-php71/root/usr/bin/php /usr/share/centreon/cron/centstorage_purge.php

go back to your log menu and find out that there're no data older than 3 days ago
https://user-images.githubusercontent.com/7352865/56137808-e350cd00-5f95-11e9-9cf7-550262423726.png

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [ ] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

<h5> Centreon team only </h5>

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests covers 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonPurgeEngine.class.php [hunk 1]
```diff
@@ -61,6 +61,8 @@ public function __construct()
             '< __RETENTION__) OR (expire_time < __RETENTION__ AND expire_time <> 0)';
         $this->purgeDowntimesQuery = 'DELETE FROM downtimes WHERE (actual_end_time is not null and actual_end_time ' .
             '< __RETENTION__) OR (deletion_time is not null and deletion_time < __RETENTION__)';
+	$this->purgeAuditLogQuery = 'DELETE FROM log_action WHERE action_log_date < __RETENTION__';
+	'NOT IN (SELECT action_log_id FROM log_action);';
```
Comment (loiclau) [inline www/class/centreonPurgeEngine.class.php]: It 's the next of the previous line?

Comment (tanguyvda) [reply www/class/centreonPurgeEngine.class.php]: thanks for pointing that out. It was there from a previous iteration. I've cleaned up the code but it looks like that i'm blind enough and missed this line.

File: www/class/centreonPurgeEngine.class.php [hunk 2]
```diff
@@ -61,6 +61,7 @@ public function __construct()
             '< __RETENTION__) OR (expire_time < __RETENTION__ AND expire_time <> 0)';
         $this->purgeDowntimesQuery = 'DELETE FROM downtimes WHERE (actual_end_time is not null and actual_end_time ' .
             '< __RETENTION__) OR (deletion_time is not null and deletion_time < __RETENTION__)';
+	$this->purgeAuditLogQuery = 'DELETE FROM log_action WHERE action_log_date < __RETENTION__';
```
Comment (sc979) [inline www/class/centreonPurgeEngine.class.php]: indentation seems odd

**Comment by sc979** [comment]
@TODO -> help message, french documentation and translation

**Comment by loiclau** [comment]
do not forget to backport the upgrade script with "if not exist'

**Comment by tanguyvda** [comment]
> audit_log_retention column is not added on fresh install

done

**Comment by tanguyvda** [comment]
> use spaces instead of tabs for indentation

my editor wasn't set at that time :')

**Comment by kduret** [comment]
replaced by https://github.com/centreon/centreon/pull/7709

**Pull request closed without merging**